python/tank/authentication/resources/build_resources.sh

```diff
@@ -17,37 +17,64 @@ if [ -z "${PYTHON_BASE}" ]; then
     PYTHON_BASE="/Applications/Shotgun.app/Contents/Resources/Python"
 fi
 
-# Remove any problematic profiles from pngs.
-for f in *.png; do mogrify $f; done
-
 # Helper functions to build UI files
 function build_qt {
-    echo " > Building " $2
-
     # compile ui to python
+    echo "$1 $2 > $UI_PYTHON_PATH/$3.py"
     $1 $2 > $UI_PYTHON_PATH/$3.py
-
-    # replace PySide imports with local imports and remove line containing Created by date
-    sed -i "" -e "s/from PySide import/from .qt_abstraction import/g" -e "/# Created:/d" $UI_PYTHON_PATH/$3.py
+    # replace PySide2 imports with local imports and remove line containing Created by date
+    sed -i"" -E "/^from PySide2.QtWidgets(\s.*)?$/d; /^\s*$/d" $UI_PYTHON_PATH/$3.py
+    sed -i"" -E "s/^(from PySide.\.)(\w*)(.*)$/from .qt_abstraction import \2\nfor name, cls in \2.__dict__.items():\n    if isinstance(cls, type): globals()[name] = cls\n/g" $UI_PYTHON_PATH/$3.py
+    sed -i"" -E "s/from PySide2 import/from .qt_abstraction import/g" $UI_PYTHON_PATH/$3.py
 }
 
 function build_ui {
-    build_qt "${PYTHON_BASE}/bin/python ${PYTHON_BASE}/bin/pyside-uic --from-imports" "$1.ui" "../ui/$1"
+    build_qt "$1 -g python --from-imports" "$2.ui" "$2"
```

Comment:
Ok so now I see that we are following this example from Flame (shotgunsoftware/tk-flame-review#17), I'm not sure it's the best syntax.

The command line to operate becomes much more complex.
Could we keep the logic with `PYTHON_BASE` or maybe another variable for the root folder where we can find the `uic` and `rcc` scripts ?

Byt the way, we ship them with the desktop app:

- Linux: /opt/Shotgun/Python3/lib/python3.10/site-packages/PySide2
- macOS: /Applications/Shotgun.app/Contents/Resources/Python3/lib/python3.10/site-packages/PySide2

So that could be great default values for the script

Reply:
Even though this [pull request #17](https://github.com/shotgunsoftware/tk-flame-review/pull/17)  was merged, it had several errors because the imports were not correct and that was fixed in [pull request #31](https://github.com/shotgunsoftware/tk-flame-review/pull/31/files?diff=split&w=0#diff-3a5749423e737b81cb65521f2192f063873c34a669a4169c20d1cb6250d8746d), that is why I followed the example of the "sed(s)" that exist there.

To be able to execute build_resources.sh it would be as recommended for tk-flame ([in last part of this wiki](https://wiki.autodesk.com/pages/viewpage.action?spaceKey=SHOT&title=Editing+Toolkit+graphic+resources) ):
`./build_resources.sh` -u "${PYENV_VIRTUAL_ENV}/bin/pyside2-uic" -r "${PYENV_VIRTUAL_ENV}/bin/pyside2-rcc" `

because I think that this way we would have an easier time being able to place any path regardless of the OS.

Reply:
I still don't understand why we need to specify 2 different path in the command line instead of just giving a python path.

Reply:
I made a change, now the build_resources.sh can be executed with only the python_path:
```
./build_resources.sh -p "C:\Users\chaucae\Documents\projects\tk-core\.venv\Scripts"
```

Reply:
Is this change optional? I mean, can it be autodetected if no `-p` argument is passed? I'm not familiar with the previous behavior.